WcaOnRails/app/webpacker/components/Results/ResultForm/DeleteResultButton.js

```diff
@@ -0,0 +1,26 @@
+import React, { useState } from 'react';
+
+import { Button, Checkbox } from 'semantic-ui-react';
+
+function DeleteResultButton({ deleteAction }) {
+  const [confirmed, setConfirmed] = useState(false);
+  return (
+    <div>
+      <Button
+        negative
+        className="delete-result-button"
+        disabled={!confirmed}
+        onClick={deleteAction}
+      >
+        Delete the result
+      </Button>
+      <Checkbox
+        label="Yes, I want to delete that result"
+        checked={confirmed}
+        onChange={() => setConfirmed((prev) => !prev)}
```

Comment:
It's not disregarded: `prev` contains the value the checkbox when `setConfirmed` is called.

The idea is to avoid the onChange function from being recreated every time the checkbox value changes (but I realized I forgot to use `useCallback`, which I've fixed).